packages/babel-types/src/converters/gatherSequenceExpressions.js

```diff
@@ -67,13 +68,16 @@ export default function gatherSequenceExpressions(
     } else if (isEmptyStatement(node)) {
       // empty statement so ensure the last item is undefined if we're last
       ensureLastUndefined = true;
```

Comment:
If _all_ of `nodes` are empty statements, it should `ensureLastUndefined`. i.e. both `eval("{ ; }")` and `eval("{ ;;; }")` returns `undefined`. Note that you can mentally think of `do { ; }` as `eval("{ ; }")`.

However, when empty statements follow any value producing statements, it should not `ensureLastUndefined`. i.e. `eval("{ 3;; }")` returns `3`.

Therefore we should check 

- If the empty statement is the _first_ of `nodes`, we should `ensureLastUndefined`, otherwise it is a no-op because it can always be omitted. Conceptually it is more like `ensureFirstUndefined`.

I don't think need to track `multiColons` because
- the specific value of multiColons is _not_ used as a number later
- Transformers operates on AST level (`EmptyStatement` v.s. `colon`) while colon is a grammar structure and should be addressed by tokenizer instead

Reply:
yup that makes sense, just have a question: if accessing the `path` isn't available to check if it's `inList` or `key` is there a way to do it with the array of `nodes` being passed to the function? 

I've also explored the `scope` that gets passed and trying to use it's `path` but `inList` and `key` are `undefined` for it.